.github/workflows/CI_dev_environment.yml

```
@@ -0,0 +1,55 @@
+name: CI Test dev environment
+
+on:
+  push:
+  pull_request:
+    types: [opened, reopened]
+
+jobs:
+
+  test_dev:
+    name: test dev environment / python-${{ matrix.python-version }} / ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+        python-version: ['3.7', '3.8']
```

I would exclude version already tested in `first_check` and `test_with_conda`
```suggestion
        python-version: ['3.7', '3.8']
        exclude:
          - os: ubuntu-latest
            python-version: 3.8 # or 3.9 when #239 is merged 
```